Random failures in test_dtm

About 1 time out of 10, I see:
<pre>
        dtm = DistanceToMeasure(k, implementation="hnsw")
        r3 = dtm.fit_transform(pts)
>       assert r3 == pytest.approx(r0)
E       assert array([0.08332776, 0.09602351, 0.11939976, 0.09936775, 0.11338955,\n       0.10259993, 0.15803374, 0.10664798, 0.170212...80446, 0.16543095, 0.15170512,\n       0.11888   , 0.12101172, 0.0895013 , 0.12953797, 0.10638626],\n      dtype=float32) == approx([0.08332777584221225 ± 8.3e-08, 0.09602351015791077 ± 9.6e-08, 0.11939976694275099 ± 1.2e-07, 0.099367757958618...01171191337208 ± 1.2e-07, 0.08950130652495468 ± 9.0e-08, 0.12953795336608637 ± 1.3e-07, 0.10638626363698155 ± 1.1e-07])
E        +  where approx([0.08332777584221225 ± 8.3e-08, 0.09602351015791077 ± 9.6e-08, 0.11939976694275099 ± 1.2e-07, 0.099367757958618...01171191337208 ± 1.2e-07, 0.08950130652495468 ± 9.0e-08, 0.12953795336608637 ± 1.3e-07, 0.10638626363698155 ± 1.1e-07]) = <function approx at 0x7f9b9ae3dca0>(array([0.08332778, 0.09602351, 0.11939977, 0.09936776, 0.11338953,\n       0.10259993, 0.15803375, 0.10664799, 0.170212...78, 0.1080068 , 0.08780445, 0.16543098, 0.15170512,\n       0.11888   , 0.12101171, 0.08950131, 0.12953795, 0.10638626]))
E        +    where <function approx at 0x7f9b9ae3dca0> = pytest.approx
</pre>

--- a/src/python/test/test_dtm.py
+++ b/src/python/test/test_dtm.py
@@ -16,7 +16,7 @@ import torch
 
 def test_dtm_compare_euclidean():
     pts = numpy.random.rand(1000, 4)
-    k = 3
+    k = 6
     dtm = DistanceToMeasure(k, implementation="ckdtree")
     r0 = dtm.fit_transform(pts)
     dtm = DistanceToMeasure(k, implementation="sklearn")
@@ -27,7 +27,7 @@ def test_dtm_compare_euclidean():
     assert r2 == pytest.approx(r0)
     dtm = DistanceToMeasure(k, implementation="hnsw")
     r3 = dtm.fit_transform(pts)
-    assert r3 == pytest.approx(r0)
+    assert r3 == pytest.approx(r0, rel=0.1)
     from scipy.spatial.distance import cdist
 
     d = cdist(pts, pts)

